pkg/network/bootstrap.go

```diff
@@ -0,0 +1,40 @@
+package network
+
+import (
+	"github.com/pkg/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gophercloud/utils/openstack/clientconfig"
+
+	operv1 "github.com/openshift/api/operator/v1"
+)
+
+type KuryrBootstrapData struct {
+	ServiceSubnet     string
+	PodSubnetpool     string
+	WorkerNodesRouter string
+	WorkerNodesSubnet string
+	PodSecurityGroups []string
+	ClusterID         string
+	OpenStackCloud    clientconfig.Cloud
+}
+
+type BootstrapData struct {
+	Kuryr KuryrBootstrapData
+}
+
+// Bootstrap creates resources required by SDN on the cloud.
+func Bootstrap(conf *operv1.NetworkSpec, client client.Client) (*BootstrapData, error) {
+	emptyData := BootstrapData{}
+	switch conf.DefaultNetwork.Type {
+	case operv1.NetworkTypeKuryr:
+		return bootstrapKuryr(conf, client)
+	case operv1.NetworkTypeOpenShiftSDN:
+		return &emptyData, nil
```

Comment:
Probably easier just to return nil?

Reply:
Right, for some reason I made all the other functions in the chain to receive bare struct. Changed them to be able to use a pointer.